From 63027c62eaa1b6c5b0d2762a511f1611b15d3728 Mon Sep 17 00:00:00 2001
From: Stephan Hartmann <stha09@googlemail.com>
Date: Mon, 19 Dec 2022 13:56:17 +0000
Subject: [PATCH] libstdc++: fix incomplete type of download::InProgressDownloadManager

Destructor of std::unique_ptr in libstdc++ uses sizeof() which
requires full definition of download::InProgressDownloadManager
for return type of content::BrowserContext::
RetrieveInProgressDownloadManager().

Bug: 957519
Change-Id: If99aa8d52238bacb1cb559a300e14ed3a05b7297
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/4111526
Reviewed-by: Bo Liu <boliu@chromium.org>
Commit-Queue: Stephan Hartmann <stha09@googlemail.com>
Cr-Commit-Position: refs/heads/main@{#1084924}
---

diff --git a/content/browser/browser_context.cc b/content/browser/browser_context.cc
index 6180969..22e9dbb 100644
--- a/content/browser/browser_context.cc
+++ b/content/browser/browser_context.cc
@@ -30,6 +30,7 @@
 #include "base/unguessable_token.h"
 #include "build/build_config.h"
 #include "build/chromeos_buildflags.h"
+#include "components/download/public/common/in_progress_download_manager.h"
 #include "components/services/storage/privileged/mojom/indexed_db_control.mojom.h"
 #include "content/browser/blob_storage/chrome_blob_storage_context.h"
 #include "content/browser/browser_context_impl.h"
